library/ssl_tls12_client.c

```diff
@@ -1764,9 +1764,11 @@ static int ssl_parse_server_ecdh_params(mbedtls_ssl_context *ssl,
         return MBEDTLS_ERR_SSL_DECODE_ERROR;
     }
 
+#if !defined(PSA_WANT_ALG_FFDH)
```

Comment:
This seems wrong: we're in a TLS 1.2 file, and only TLS 1.3 should be influenced by `PSA_WANT_ALG_FFDH`. However perhaps this is related to the issue about how fields are guarded, which I agreed can be considered out of scope.

Reply:
Tracked in https://github.com/Mbed-TLS/mbedtls/issues/7612#issuecomment-1568184319 so marking as resolved here.